src/pudl/workspace/zenodo_download.py

```diff
@@ -0,0 +1,336 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import argparse
+import os
+import logging
+import re
+import requests
+import sys
+import yaml
+
+
+class ZenodoDownload:
+    """
+    Handle connections and downloading of Zenodo Source archives
+    """
+
+    def __init__(self, loglevel="WARNING", verbose=False, sandbox=False):
+        """
+        Create a ZenDownloader instance
+
+        Args:
+            cfg: dict of the config from ~/.pudl.yml
+            loglevel: str, logging level
+            verbose: boolean. If true, logs printed to stdout
+            sandbox: boolean. If true, use the sandbox server instead of
+                     production
+
+        Returns:
+            ZenDownloader instance
+        """
+        logger = logging.Logger(__name__)
+        logger.setLevel(loglevel)
+
+        if verbose:
+            logger.addHandler(logging.StreamHandler())
+
+        self.logger = logger
+        logger.info("Logging at loglevel %s" % loglevel)
+
+        cfg = self.load_config()
+
+        if sandbox:
+            self.api_root = "https://sandbox.zenodo.org/api"
+            config_root = cfg["zenodo_download"]["sandbox"]
+        else:
+            self.api_root = "https://zenodo.org/api"
+            config_root = cfg["zenodo_download"]["production"]
+
+        self.token = os.environ.get("ZENODO_TOKEN", config_root["token"])
+        self.dois = config_root["dois"]
+
+        self.output_root = os.environ.get("PUDL_IN", cfg["pudl_in"])
+
+    @staticmethod
+    def load_config():
+        """
+        Load ~/.pudl.yml as a dict.  Raise error if there is no
+        zenodo_download section.
+
+        Args: None
+        Returns:
+            dict of configuration parameters
+        """
+        with open(os.path.expanduser("~/.pudl.yml"), "r") as f:
+            cfg = yaml.load(f.read(), Loader=yaml.FullLoader)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of unsafe yaml load. Allows instantiation of arbitrary objects. Consider yaml.safe_load().](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=5554119)